matlab/src/cpp/arrow/matlab/type/proxy/field.cc

```diff
@@ -0,0 +1,129 @@
+                return std::make_shared<type_proxy::PrimitiveCType<int16_t>>(std::static_pointer_cast<arrow::Int16Type>(datatype));
+            case arrow_type::INT32:
+                return std::make_shared<type_proxy::PrimitiveCType<int32_t>>(std::static_pointer_cast<arrow::Int32Type>(datatype));
+            case arrow_type::INT64:
+                return std::make_shared<type_proxy::PrimitiveCType<int64_t>>(std::static_pointer_cast<arrow::Int64Type>(datatype));
+            case arrow_type::FLOAT:
+                return std::make_shared<type_proxy::PrimitiveCType<float>>(std::static_pointer_cast<arrow::FloatType>(datatype));
+            case arrow_type::DOUBLE:
+                return std::make_shared<type_proxy::PrimitiveCType<double>>(std::static_pointer_cast<arrow::DoubleType>(datatype));
+            case arrow_type::BOOL:
+                return std::make_shared<type_proxy::PrimitiveCType<bool>>(std::static_pointer_cast<arrow::BooleanType>(datatype));
+            case arrow_type::STRING:
+                return std::make_shared<type_proxy::StringType>(std::static_pointer_cast<arrow::StringType>(datatype));
+            case arrow_type::TIMESTAMP:
+                return std::make_shared<type_proxy::TimestampType>(std::static_pointer_cast<arrow::TimestampType>(datatype));
+            default:
+                return arrow::Status::Invalid("Unsupported DataType: " + datatype->ToString());
+        }
+    }
+
+
+    void Field::type(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+
+        auto datatype = field->type();
+        MATLAB_ASSIGN_OR_ERROR_WITH_CONTEXT(auto proxy, makeTypeProxy(datatype), context, "arrow:field:FailedToCreateTypeProxy");
+        const auto proxy_id = libmexclass::proxy::ProxyManager::manageProxy(proxy);
+
+        mda::ArrayFactory factory;
+        context.outputs[0] = factory.createScalar(proxy_id);
+        context.outputs[1] = factory.createScalar(static_cast<uint64_t>(datatype->id()));
+    }
+
+    void Field::toString(libmexclass::proxy::method::Context& context) {
+        namespace mda = ::matlab::data;
+        mda::ArrayFactory factory;
+
+        const auto str_utf8 = field->ToString();
+        MATLAB_ASSIGN_OR_ERROR_WITH_CONTEXT(const auto str_utf16, arrow::util::UTF8StringToUTF16(str_utf8), context, error::UNICODE_CONVERSION_ERROR_ID);
+        auto str_mda = factory.createScalar(str_utf16);
+        context.outputs[0] = str_mda;
+    }
+
+    libmexclass::proxy::MakeResult Field::make(const libmexclass::proxy::FunctionArguments& constructor_arguments) {
+        namespace mda = ::matlab::data;
+        using FieldProxy = arrow::matlab::type::proxy::Field;
+
+        mda::StructArray opts = constructor_arguments[0];
+        const mda::StringArray name_mda = opts[0]["Name"];
+        const mda::TypedArray<uint64_t> type_proxy_id_mda = opts[0]["TypeProxyID"];
+
+        const std::u16string& utf16_name = name_mda[0];
```

Comment:
We may want to use `name_utf16` as variable name before other codes use `str_utf16`.

Reply:
Fixed.